wlauto/workloads/blogbench/__init__.py

```diff
@@ -0,0 +1,78 @@
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# pylint: disable=attribute-defined-outside-init
+import os
+
+from wlauto import Workload, Parameter, Executable
+
+
+class BlogbenchWorkload(Workload):
+
+    name = 'blogbench'
+    description = """
+    Blogbench is a portable filesystem benchmark that tries to reproduce the
+    load of a real-world busy file server.
+
+    Blogbench stresses the filesystem with multiple threads performing random
+    reads, writes and rewrites in order to get a realistic idea of the
+    scalability and the concurrency a system can handle.
+
+    Source code are available from:
+        https://download.pureftpd.org/pub/blogbench/
+    """
+
+    parameters = [
+        Parameter('iterations', kind=int, default=30,
+                  description='The number of iterations to run'),
+        Parameter('directory', kind=str,
+                  default='/mnt/blogbench',
+                  description='Specify an empty and writable directory')
+    ]
+
+    def initialize(self, context):
+        host_exe = context.resolver.get(Executable(self, self.device.abi,
+                                                   'blogbench'))
+        BlogbenchWorkload.binary = self.device.install(host_exe)
+
+    def setup(self, context):
+        # Total test duration equal to iteration*frequency
+        # The default frequency is 10 seconds, plus 5 here as a buffer.
+        self.timeout = self.iterations * 15
+        self.device.execute('rm -rf {}'.format(self.directory), timeout=60)
+        self.device.execute('mkdir -p {}'.format(self.directory))
+        self.results = os.path.join(self.device.working_directory,
+                                    'blogbench.output')
+        self.command = ('{} --iterations {} --directory {} > {}'
+                        .format(self.binary, self.iterations, self.directory,
+                                self.results))
+
+    def run(self, context):
+        self.output = self.device.execute(self.command, timeout=self.timeout)
+
+    def update_result(self, context):
+        host_file = os.path.join(context.output_directory, 'blogbench.output')
+        self.device.pull_file(self.results, host_file)
+
+        with open(host_file, 'r') as fh:
+            for line in fh:
+                if any('Final score for ' + x in line
+                       for x in ['writes', 'reads']):
+                    line = line.split(':')
+                    metric = line[0].lower().strip().replace(' ', '_')
+                    score = int(line[1].strip())
+                    context.result.add_metric(metric, score, 'blogs')
+
+    def finalize(self, context):
+        self.device.uninstall(self.binary)
```

Comment:
We don't usually remove binaries from the device, is there a reason you need to for this workload?